include/revng/Model/VerifyHelper.h

```diff
@@ -144,9 +145,11 @@ class VerifyHelper {
   }
 
 public:
-  [[nodiscard]] bool isGlobalSymbol(const model::Identifier &Name) const;
-  [[nodiscard]] bool registerGlobalSymbol(const model::Identifier &Name,
-                                          const std::string &Path);
+  [[nodiscard]] bool isGlobalSymbol(std::string_view Name) const;
+  [[nodiscard]] bool registerNamingHelper(const model::Binary &Binary);
```

Comment:
As mentioned elsewhere, would it make sense to have `model::Binary` be an argument of `VerifyHelper` itself?

Reply:
It was at one point. But the thing is, it would still need a separate `NamingHelper`. So we kind of don't win anything - hence why I get rid of it.

Reply:
Can you elaborate?
If it's in the constructor, we can avoid having the `registerNamingHelper` method and avoid `VerifyHelper::NamingHelper` to be a `unique_ptr`.

Reply:
fuck, I misread it! All good, building the VH from the binary works!

Reply:
The constructor worked great!

The only issue is that even this didn't let us avoid a pointer in here. There was one more reason for it I overlooked initially: this header cannot include `NameBuilder.h` because `NameBuilder.h` has to include all the different headers (`Function.h`, `TypeDefinition.h` and so on) that include this header - leading to a cycle.

Reply:
What pointer survived? Please give me a specific reference.

Reply:
The fact that the `NameBuilder` member is a `unique_ptr` inside the `VerifyHelper`.

Reply:
> NameBuilder.h has to include all the different headers (Function.h, TypeDefinition.h and so on)

Does it? It seems to me that you could forward declare most of the things and outline implementations.

Reply:
I guess it can be done, but then we're losing some inlining.

But even putting that aside, I feel like it's weird to dictate the requirements on a widely-used interface based on a verification routine. `VerifyHelper` feels pretty "internal, for the lack of a better word. I'd rather it be ugly, than the interface we get all of our names from.

Reply:
OK, I'll leave the decision to you.

If you keep `unique_ptr`, add a thorough comment describing why we need that and what would need to be done in order to switch to a regular instance.

Reply:
It was even deeper than expected: even `model::Identifier` depends on the `model::VerifyHelper`, so yeah, the other way around would be very awkward.

Comment I added:
```cpp
  // Name builder is stored as a pointer in order to avoid an inclusion loop:
  // NameBuilder methods return `model::Identifier`, which depends on this
  // helper.
  std::unique_ptr<model::NameBuilder> NameBuilder = nullptr;
```